srv/salt/_modules/osd.py

```diff
@@ -1046,94 +1050,53 @@ def _filestore_args(self):
         args = ""
-        if self.osd.wal and self.osd.db:
-            if self.is_partitioned(self.osd.wal):
+        if not self.osd.encryption:
+
+            # WAL cornercase with sizes
+            if self.osd.wal and self.osd.wal_size and self.osd.wal != self.osd.device:
                 partition = self.highest_partition(self.osd.wal, 'wal')
                 if partition:
-                    args = "--block.wal {}{} ".format(self.osd.wal, partition)
+                    args += "--block.wal {}{} ".format(self.osd.wal, partition)
                 else:
-                    args = "--block.wal {} ".format(self.osd.wal)
-            else:
-                args = "--block.wal {} ".format(self.osd.wal)
+                    args += "--block.wal {} ".format(self.osd.wal)
 
-            if self.is_partitioned(self.osd.db):
-                partition = self.highest_partition(self.osd.db, 'db')
+            # DB cornercase with sizes
+            if self.osd.db and self.osd.db_size and self.osd.db != self.osd.device:
+                partition = self.highest_partition(self.osd.wal, 'db')
                 if partition:
                     args += "--block.db {}{} ".format(self.osd.db, partition)
                 else:
                     args += "--block.db {} ".format(self.osd.db)
-            else:
+
+            # Generic case withouth sizes 
+            if self.osd.wal and not (self.osd.wal_size or self.osd.db_size):
+                args += "--block.wal {} ".format(self.osd.wal)
+            if self.osd.db and not (self.osd.wal_size or self.osd.db_size):
                 args += "--block.db {} ".format(self.osd.db)
-        else:
-            if self.osd.wal:
-                log.debug("wal: {}".format(self.osd.wal))
-                if self.is_partitioned(self.osd.wal):
-                    partition = self.highest_partition(self.osd.wal, 'wal')
-                    if partition:
-                        args += "--block.wal {}{} ".format(self.osd.wal, partition)
-                    else:
-                        args += "--block.wal {} ".format(self.osd.wal)
-                    log.debug("args: {}".format(args))
 
-                    partition = self.highest_partition(self.osd.wal, 'db')
-                    if partition:
-                        args += "--block.db {}{} ".format(self.osd.wal, partition)
-                    else:
-                        args += "--block.db {} ".format(self.osd.wal)
-                    log.debug("args: {}".format(args))
-                else:
-                    if self.osd.wal == self.osd.device:
-                        log.warn("Separate wal partition on {} unnecessary - triggers ceph-disk bug".format(self.osd.wal))
-                    else:
-                        args += "--block.wal {} --block.db {} ".format(self.osd.wal, self.osd.wal)
+        if self.osd.encryption and (self.osd.wal_size or self.osd.db_size):
+            log.warn("""Your specified wal/db_sizes will not be respected. 
```

Comment:
nit: s/honored/respected, not a big deal though

Reply:
Could it be written this way:

```
The --wal-size and --db-size options are not supported for encrypted OSDs, so
the values you specified will be ignored. Please specify the WAL and DB sizes
via ceph.conf with:
                           bluestore block db size = size
                           bluestore block wal size = size
```

Reply:
much better :)

Reply:
updated, thanks